coalib/bears/Bear.py

```diff
@@ -288,6 +335,20 @@ def run_bear_from_section(self, args, kwargs):
                 self.name), str(err))
             return
 
+        debug_bears = kwargs.get('debug_bears')
+        if debug_bears is None:
+            return self.run(*args, **kwargs)
+        elif debug_bears is False:
+            kwargs.pop('debug_bears')
+            return self.run(*args, **kwargs)
+        elif (str(debug_bears[0]).lower().strip())=='true' or \
+            (str(self.__class__.__name__) in debug_bears):
+            self.debugger_config(*args,**kwargs)
+            kwargs.pop('debug_bears')
+        else:
+            kwargs.pop('debug_bears')
+            logging.info("Passed the Bear name as a argument to debug")
+            return self.run(*args, **kwargs)
```

Comment:
Instead of fiddling here with sections, it makes sense to make the `__init__` function accept a `debug` parameter, that controls debugging per bear. Then you would shift this logic to the bear-instantiation part of the core (then you check the section for debug configs, and instantiate bears accordingly in debug mode or not).

Reply:
`debug` is just a so essential feature, that it imo should reside in `__init__`^^ Would be also easier to port this to the NextGen-Core :+1:

Reply:
Done till noon I will try to rewrite the `LInter` part then update the PR for review